### PR TITLE
test: Enable Acceptance test in CI for `mongodbatlas_federated_settings_identity_provider`

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Enable acceptance test for federated_settings_identity_provider in the CI pipeline
Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-220799

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/acceptance-tests-runner.yml [hunk 1]
```diff
@@ -607,4 +634,34 @@ jobs:
...
+          MONGODB_ATLAS_FEDERATION_SETTINGS_ID: ${{ inputs.mongo_atlas_federation_settings_id }}
+          MONGODB_ATLAS_FEDERATED_OKTA_IDP_ID: ${{ inputs.mongo_atlas_federatied_okta_idp_id }}
+          MONGODB_ATLAS_FEDERATED_IDP_ID: ${{ inputs.mongo_atlas_federatied_idp_id }}
+          MONGODB_ATLAS_FEDERATED_OIDC_IDP_ID: ${{ inputs.mongo_atlas_federatied_oidc_idp_id }}
+          MONGODB_ATLAS_FEDERATED_SSO_URL: ${{ inputs.mongo_atlas_federatied_sso_url }}
+          MONGODB_ATLAS_FEDERATED_ISSUER_URI: ${{ inputs.mongo_atlas_federatied_issuer_uri }}
+          MONGODB_ATLAS_FEDERATED_PROJECT_ID: ${{ inputs.mongo_atlas_federatied_project_id }}
+          MONGODB_ATLAS_FEDERATED_ORG_ID: ${{ inputs.mongo_atlas_federatied_org_id }}
+          TEST_REGEX: "^TestAccFederatedSettingsIdentityProvider"
+        run: make testacc
```
Comment (lantoli) [inline .github/workflows/acceptance-tests-runner.yml]: finish last line with newline

File: internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider_test.go [hunk 2]
```diff
@@ -29,9 +28,9 @@ func TestAccFedDSFederatedSettingsIdentityProvider_samlBasic(t *testing.T) {
 					resource.TestCheckResourceAttrSet(resourceName, "associated_orgs.#"),
 					resource.TestCheckResourceAttrSet(resourceName, "acs_url"),
 					resource.TestCheckResourceAttrSet(resourceName, "display_name"),
```
Comment (lantoli) [inline internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider_test.go]: resource.TestCheckResourceAttrSet(resourceName, "display_name"),
is redundandt if TestCheckResourceAttr is used later to check the value

**Review by lantoli — APPROVED** [review]

File: internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider_test.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider_test.go]: Done

File: .github/workflows/acceptance-tests-runner.yml (hunk 1 above)
Comment (oarbusi) [reply .github/workflows/acceptance-tests-runner.yml]: Done

**Comment by lantoli** [comment]
in PR description just need to say CLOUDP-220799, not the full url/link

**Review by lantoli — APPROVED** [review]

File: .github/workflows/acceptance-tests.yml [hunk 3]
```diff
@@ -68,3 +68,11 @@ jobs:
       azure_service_principal_id: ${{ inputs.atlas_cloud_env == 'qa' && vars.AZURE_SERVICE_PRINCIPAL_ID_QA || vars.AZURE_SERVICE_PRINCIPAL_ID }}
       azure_tenant_id: ${{ inputs.atlas_cloud_env == 'qa' && vars.AZURE_TENANT_ID_QA || vars.AZURE_TENANT_ID }}
       aws_secret_name: ${{ inputs.atlas_cloud_env == 'qa' && vars.AWS_SECRET_NAME_QA || vars.AWS_SECRET_NAME }}
+      mongo_atlas_federation_settings_id: ${{ inputs.atlas_cloud_env == 'qa' && vars.MONGODB_ATLAS_FEDERATION_SETTINGS_ID_QA || vars.MONGODB_ATLAS_FEDERATION_SETTINGS_ID }}
+      mongo_atlas_federatied_okta_idp_id: ${{ inputs.atlas_cloud_env == 'qa' && vars.MONGODB_ATLAS_FEDERATED_OKTA_IDP_ID_QA || vars.MONGODB_ATLAS_FEDERATED_OKTA_IDP_ID }}
```
Comment (AgustinBettati) [inline .github/workflows/acceptance-tests.yml]: typo is present in other paramters as well
```suggestion
      mongodb_atlas_federated_okta_idp_id: ${{ inputs.atlas_cloud_env == 'qa' && vars.MONGODB_ATLAS_FEDERATED_OKTA_IDP_ID_QA || vars.MONGODB_ATLAS_FEDERATED_OKTA_IDP_ID }}
```

**Review by AgustinBettati — COMMENTED** [review]
great to have this running in the CI 💯

File: .github/workflows/acceptance-tests.yml (hunk 3 above)
Comment (oarbusi) [reply .github/workflows/acceptance-tests.yml]: good catch! Changed

**Pull request merged**